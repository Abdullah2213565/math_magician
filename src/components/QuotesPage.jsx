import React from 'react';

const QuotesPage = () => (
  <div className="quotes-page">
    <h3>
    Without mathematics, thereâ€™s nothing you can do.
      <br />
      Everything around you is mathematics. Albert Einstein
    </h3>
  </div>
);

export default QuotesPage;