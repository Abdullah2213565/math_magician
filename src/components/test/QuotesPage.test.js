import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import QuotesPage from '../QuotesPage';

describe('Test the quote component', () => {
  test('renders maths quote', () => {
    render(<QuotesPage />);
    const quote = screen.getByText(/Without mathematics, thereâ€™s nothing you can do/i);
    expect(quote).toBeInTheDocument();
  });

  test('Quotes is rendered', () => {
    const quote = renderer
      .create(<QuotesPage />)
      .toJSON();
    expect(quote).toMatchSnapshot();
  });
});
