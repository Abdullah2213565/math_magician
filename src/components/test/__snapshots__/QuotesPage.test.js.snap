// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the quote component Quotes is rendered 1`] = `
<div
  className="quotes-page"
>
  <h3>
    Without mathematics, thereâ€™s nothing you can do.
    <br />
    Everything around you is mathematics. Albert Einstein
  </h3>
</div>
`;
